src/Sarif/Core/Result.cs

```diff
@@ -93,6 +93,23 @@ public ReportingDescriptor GetRule(Run run = null)
             return new ReportingDescriptor() { Id = this.RuleId ?? this.Rule?.Id };
         }
 
+        public bool IsSuppressed()
+        {
+            if (this == null)
+            {
+                throw new ArgumentNullException(nameof(Result));
+            }
+
+            IList<Suppression> suppressions = this.Suppressions;
+            if (suppressions == null || suppressions.Count == 0)
```

Comment:
>suppressions == null [](start = 16, length = 20)

if suppressions is null, we don't have *any* suppressions related data in this log and so we can't answer the question. Two ideas, either throw an exception for this case of use bool TryIsSuppressed...

Reply:
updated. One thing that i changed: we won't throw exception. What will happen is if the Result isn't available, then we will return false.

---
In reply to: [529072719](https://github.com/microsoft/sarif-sdk/pull/2159#discussion_r529072719) [](ancestors = 529072719)

Reply:
that's right! in the TryXXX pattern, this is exactly the behavior you want. 

---
In reply to: [529483550](https://github.com/microsoft/sarif-sdk/pull/2159#discussion_r529483550) [](ancestors = 529483550,529072719)